src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/TextEditingTargetWidget.java

```diff
@@ -116,13 +124,88 @@ public void removeFindReplace(FindReplaceBar findReplaceBar)
             } 
          });
       
-      panel_ = new PanelWithToolbars(toolbar_ = createToolbar(fileType),
-                                    editor.asWidget(),
-                                    statusBar_);
+      editorPanel_ = new DockLayoutPanel(Unit.PX);
+      docOutlineWidget_ = new DocumentOutlineWidget(target);
+      
+      editorPanel_.addEast(docOutlineWidget_, 0);
+      editorPanel_.add(editor.asWidget());
+      
+      // Need to set the widget size deferred to ensure the docUpdateSentinel
+      // has been initialized as well
+      new Timer()
```

Comment:
Thanks -- I agree 1. is the best approach (I had missed that the sentinel was initialized right after the widget) and so moved the initialization of size to there.